js/ui/appDisplay.js

```diff
@@ -173,13 +173,121 @@ const BaseAppView = new Lang.Class({
         // Nothing by default
     },
 
+    _createItemForId: function(itemId) {
+        if (IconGridLayout.layout.iconIsFolder(itemId))
+            return Shell.DesktopDirInfo.new(itemId);
+
+        return Shell.AppSystem.get_default().lookup_app(itemId);
+    },
+
+    _createItemIcon: function(item) {
+        throw new Error('Not implemented');
+    },
+
+    getLayoutIds: function() {
+        let viewId = this.getViewId();
+        return IconGridLayout.layout.getIcons(viewId).slice();
+    },
+
+    _trimInvisible: function(ids) {
+        let appSystem = Shell.AppSystem.get_default();
+        return ids.filter((id) => {
+            return IconGridLayout.layout.iconIsFolder(id) || appSystem.lookup_app(id) || (id == EOS_APP_CENTER_ID);
+        });
+    },
+
+    _findIconChanges: function() {
+        let oldItemLayout = this._allItems.map(function(icon) { return icon.getId(); });
+        let newItemLayout = this.getLayoutIds();
+        newItemLayout = this._trimInvisible(newItemLayout);
+
+        let movedList = {};
```

Comment:
Using a `Hash.Map()` should make the code that uses the moved list a bit easier

Reply:
I think you mean JS's Map, right? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map